client/src/util/stateManager/colorHelpers.ts

```diff
@@ -186,55 +199,65 @@ function _createUserColors(
   // See createColorsByCategoricalMetadata() for another example.
   const { categories } = schema.annotations.obsByName[colorAccessor];
   const categoryMap = new Map();
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  categories.forEach((label: any, idx: any) => categoryMap.set(idx, label));
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  const scale = (idx: any) => scaleByLabel(categoryMap.get(idx));
+
+  categories?.forEach((label, idx) => categoryMap.set(idx, label));
+
+  const scale = (idx: number) => scaleByLabel(categoryMap.get(idx));
 
   return { rgb, scale };
 }
 const createUserColors = memoize(_createUserColors);
 
+interface CategoryColors {
+  [category: string]: [number, number, number];
+}
+
 function _createColorsByCategoricalMetadata(
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  data: any,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  colorAccessor: any,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  schema: any
-) {
+  data: DataframeValueArray,
+  colorAccessor: LabelType,
+  schema: Schema
+): Colors {
   const { categories } = schema.annotations.obsByName[colorAccessor];
 
   const scale = d3
     .scaleSequential(interpolateRainbow)
-    .domain([0, categories.length]);
+    .domain([0, categories?.length || 0]);
 
   /* pre-create colors - much faster than doing it for each obs */
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  const colors = categories.reduce((acc: any, cat: any, idx: any) => {
-    acc[cat] = parseRGB(scale(idx));
+  const colors = categories?.reduce((acc: CategoryColors, cat, idx) => {
+    acc[cat as string] = parseRGB(scale(idx));
     return acc;
   }, {});
 
   const rgb = createRgbArray(data, colors);
+
   return { rgb, scale };
 }
+
 const createColorsByCategoricalMetadata = memoize(
   _createColorsByCategoricalMetadata
 );
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-function createRgbArray(data: any, colors: any) {
+function createRgbArray(data: DataframeValueArray, colors?: CategoryColors) {
   const rgb = new Array(data.length);
+
+  if (!colors) {
+    throw new Error("`colors` is undefined");
+  }
+
```

Comment:
What case were we seeing `createRgbArray` called with a single arg?

Reply:
Great question! `colors` is optional here because callsite could be undefined as `colors`. E.g., https://github.com/chanzuckerberg/single-cell-explorer/pull/19/files/bfaa6ac79d14925936517da37631bc6700bf5538#r705659559